src/torrent_info.cpp

```diff
@@ -967,8 +966,6 @@ namespace libtorrent
 
 	void torrent_info::unload()
 	{
-		TORRENT_ASSERT(m_info_section.unique());
```

Comment:
I think this should probably be a shared_ptr still. you could make it a `std::shared_ptr<char[]>` still though